src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const lastElement = this[this.length - 1];
+    let string = '';
+
+    for (const element of this) {
+      const elem = element !== null && element !== undefined
```

Comment:
I think you can just write 
```js
const elem = element != null
  ? ...
  : ...;

Reply:
Tried that. You need to explicitly check for undefined as well